### PR TITLE
initial blueprint4 upgrade

### LINKED ISSUE #1395: Blueprintjs v4 upgrade
https://github.com/palantir/blueprint/wiki/Blueprint-4.0

aiming carta >= v3.0

### PR DESCRIPTION
**Pull request by veggiesaurus**
**Description**
Closes #1395. It may also close a number of table-related issues. 
Requires a fair amount of testing, and will likely break e2e tests. 

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] changelog updated / ~no changelog update needed~
- [x] ~unit test added (for functions with no dependencies)~
- [x] ~API documentation added (for public variables and methods in stores)~

For dependencies:
- [x] ~e2e test passing~ / corresponding fix added / ~new e2e test created~
- [x] ~protobuf version bumped~ / no protobuf version bumped needed
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] ~corresponding ICD test fix added (`BackendService` changed)~ / no ICD test fix needed (`BackendService` unchanged)
- [x] ~user manual prepared (for large new features)~

### CODE REVIEW

**Comment by veggiesaurus** [comment]
Assigning this to @kswang1029, as it is primarily a user-facing thing. Most of the upgrades were fairly straightforward. I've removed deprecated code where possible.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by kswang1029** [comment]
@veggiesaurus not sure why but I am seeing e2e test failures due to changes of xpath (eg. the file filter text field). I will need more time to investigate this. And you are right about the color theme: the contrast is increased and the color theme appears less warm.

**Comment by kswang1029** [comment]
One issue: the initial file browser highlight box is squeezed. 
with this PR
Screen Shot 2022-09-01 at 20 56 31
current dev
Screen Shot 2022-09-01 at 20 56 54

**Comment by kswang1029** [comment]
in the file browser, the sorting function in the the Filename, Type, Size, and Date columns does not work anymore.

**Comment by kswang1029** [comment]
@veggiesaurus @YuHsuan-Hwang there are a number of issues due to this library upgrade. I will summarize them in a google doc, just for better readability than posting here as a comment thread.

**Comment by kswang1029** [comment]
@veggiesaurus please refer to https://docs.google.com/document/d/1UgbYN7CzW2ziPeYo7bn_vDxHrSgs1ri6QDjXS86ylYk/edit?usp=sharing for a list of issues I observed with this blueprintjsV4 upgrade. Basically there are major issues with 
- spinbox
- html table
- blueprint table 🤦🏻‍♂️

**Comment by kswang1029** [comment]
@TienHao most of the issues in https://docs.google.com/document/d/1UgbYN7CzW2ziPeYo7bn_vDxHrSgs1ri6QDjXS86ylYk/edit are addressed. 👍 
there are some new others. Please refer to the above doc.

**Comment by YuHsuan-Hwang** [comment]
Upgrade to blueprint v5 in the next release (v5-beta).

File: src/App.scss [hunk 1]
```diff
@@ -58,11 +58,17 @@ body {
...
 
 // Workaround for BlueprintJS bug
-.bp3-dark .bp3-button-group:not(.bp3-minimal) > .bp3-popover-wrapper:not(:last-child) .bp3-button,
-.bp3-dark .bp3-button-group:not(.bp3-minimal) > .bp3-button:not(:last-child) {
+.#{$ns}-dark .#{$ns}-button-group:not(.#{$ns}-minimal) > .#{$ns}-popover-wrapper:not(:last-child) .#{$ns}-button,
+.#{$ns}-dark .#{$ns}-button-group:not(.#{$ns}-minimal) > .#{$ns}-button:not(:last-child) {
     margin-right: -1px !important;
 }
 
+// override menu item style with BlueprintJS v3 style
```
Comment (YuHsuan-Hwang) [inline src/App.scss]: Remove comments.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes look nice. Thanks for all the hard work!

One minor issue: the fitting dialog right-bottom toolbar position needs adjustment.

File: src/App.scss (hunk 1 above)
Comment (TienHao) [reply src/App.scss]: removed!

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
Well done @TienHao 👍

**Review by YuHsuan-Hwang — APPROVED** [review]
New changes look good.

**Review by kswang1029 — APPROVED** [review]

**Pull request merged**